pkg/controller/util.go

```diff
@@ -326,3 +329,297 @@ func addToMap(m1, m2 map[string]string) map[string]string {
 	}
 	return m1
 }
+
+// returns the CloneRequest string which contains the pvc name (and namespace) from which we want to clone the image.
+func getCloneRequestPVC(pvc *v1.PersistentVolumeClaim) (string, error) {
+	cr, found := pvc.Annotations[AnnCloneRequest]
+	if !found || cr == "" {
+		verb := "empty"
+		if !found {
+			verb = "missing"
+		}
+		return cr, errors.Errorf("annotation %q in pvc \"%s/%s\" is %s\n", AnnCloneRequest, pvc.Namespace, pvc.Name, verb)
+	}
+	return cr, nil
+}
+
+func CreateCloneSourcePod(client kubernetes.Interface, image, verbose, pullPolicy, cr, secretName string, pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	strArr := strings.Split(cr, "/")
+    var ns string = ""
+	if strArr == nil || len(strArr) < 2 {
+		glog.V(Vdebug).Infof("Wrong CloneRequest Annotation")
+		return nil, nil
+	}
+	ns = strArr[0]
+	pvcName := strArr[1]
+	
+	pod := MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, secretName, pvcName)
+
+	pod, err := client.CoreV1().Pods(ns).Create(pod)
+	if err != nil {
+		return nil, errors.Wrap(err, "source pod API create errored")
+	}
+	glog.V(Vuser).Infof("source pod \"%s/%s\" (image: %q) created\n", pod.Namespace, pod.Name, image)
+	return pod, nil
+}
+
+// return the clone source pod spec based on the target pvc.
+func MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, secret string, pvcName string) *v1.Pod {
+	// source pod name contains the pvc name
+	podName := fmt.Sprintf("%s-", CLONER_SOURCE_PODNAME)
+
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: podName,
+			Annotations: map[string]string{
+				AnnCreatedBy: "yes",
+			},
+		},
+		Spec: v1.PodSpec{
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
```

Comment:
you don't care about performance in case we have many cloning pods?

Reply:
It would be the same situation as using node affinity as the expectation is that only one node is labeled for host-assisted cloning.  We could actually fix this better by using a dynamic app label for each pair of cloning pods.  Basically label is host-assisted-clone-<hash value> and use that for the source and target.  Each pair would get a different label and thus could be scheduled on different nodes.

Reply:
works. Thank you.

Reply:
+1. A combination of pod affinity to get the coupled pair of cloning pods on the same node, with a preference to not be co-located with other cloning pods (or imported pods for that matter).

Reply:
@aglitke : only one pair of pods (first source and target) can run on the node. All other pods cannot be schedule on the same pod, when I use PodAffinity. 
However, with NodeAffinity, all of them scheduled on the same node.
Any clue, how to solve this?

Reply:
Is this the issue with using a hard-coded host path for the socket?  I guess you should dynamically generate the path ie...

```
/tmp/cdi-host-clone/<uuid-same as on the pod labels>/...
```